Build of LAPACKPP fails

Trying to build TTG fails on current master:

```
[1/426] Building CXX object _deps/lapackpp-build/CMakeFiles/lapackpp.dir/src/bbcsd.cc.o
FAILED: _deps/lapackpp-build/CMakeFiles/lapackpp.dir/src/bbcsd.cc.o 
/usr/bin/g++-10  -DLAPACKPP_ID=\"13301a1\" -I_deps/lapackpp-build/include -I_deps/lapackpp-src/include -I_deps/blaspp-build/include -I_deps/blaspp-src/include -g -fPIC   -fdiagnostics-color=always -Wall -fopenmp -std=c++17 -MD -MT _deps/lapackpp-build/CMakeFiles/lapackpp.dir/src/bbcsd.cc.o -MF _deps/lapackpp-build/CMakeFiles/lapackpp.dir/src/bbcsd.cc.o.d -o _deps/lapackpp-build/CMakeFiles/lapackpp.dir/src/bbcsd.cc.o -c _deps/lapackpp-src/src/bbcsd.cc
In file included from _deps/lapackpp-src/include/lapack/fortran.h:10,
                 from _deps/lapackpp-src/src/bbcsd.cc:7:
_deps/lapackpp-src/include/lapack/mangling.h:26:10: error: #error "One of LAPACK_FORTRAN_ADD_, LAPACK_FORTRAN_LOWER, or LAPACK_FORTRAN_UPPER must be defined to set how Fortran functions are name mangled."
   26 |         #error "One of LAPACK_FORTRAN_ADD_, LAPACK_FORTRAN_LOWER, or LAPACK_FORTRAN_UPPER must be defined to set how Fortran functions are name mangled."
      |          ^~~~~
In file included from _deps/lapackpp-src/src/bbcsd.cc:7:
_deps/lapackpp-src/include/lapack/fortran.h:48:23: error: variable or field ‘LAPACK_GLOBAL’ declared void
   48 | #define LAPACK_sgetrf LAPACK_GLOBAL(sgetrf,SGETRF)
      |                       ^~~~~~~~~~~~~
_deps/lapackpp-src/include/lapack/fortran.h:49:6: note: in expansion of macro ‘LAPACK_sgetrf’
   49 | void LAPACK_sgetrf(
      |      ^~~~~~~~~~~~~
_deps/lapackpp-src/include/lapack/fortran.h:48:37: error: ‘sgetrf’ was not declared in this scope 
   48 | #define LAPACK_sgetrf LAPACK_GLOBAL(sgetrf,SGETRF)
      |                                     ^~~~~~
_deps/lapackpp-src/include/lapack/fortran.h:49:6: note: in expansion of macro ‘LAPACK_sgetrf’
   49 | void LAPACK_sgetrf(
      |      ^~~~~~~~~~~~~
_deps/lapackpp-src/include/lapack/fortran.h:48:44: error: ‘SGETRF’ was not declared in this scope 
   48 | #define LAPACK_sgetrf LAPACK_GLOBAL(sgetrf,SGETRF)
      |                                            ^~~~~~
```

CMake invocation and log:

```
$ cmake .. -DBUILD_SHARED_LIBS=OFF -DBLA_VENDOR=Intel10_64lp_seq -DBUILD_SHARED_LIBS=OFF -Duse_cuda=OFF -DCMAKE_BUILD_TYPE=Debug -DDPLASMA_ROOT_DIR=$HOME/opt/dplasma-termdet -DPaRSEC_ROOT_DIR=$HOME/opt/parsec-termdet/ -DCMAKE_C_COMPILER=gcc-10 -DCMAKE_CXX_COMPILER=g++-10 -DCMAKE_Fortran_COMPILER=g++-10 -G Ninja
-- The CXX compiler identification is GNU 10.3.0
-- Check for working CXX compiler: /usr/bin/g++-10
-- Check for working CXX compiler: /usr/bin/g++-10 -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- EIGEN3_FOUND=1
-- Found Boost at /usr/lib/x86_64-linux-gnu/cmake/Boost-1.71.0/BoostConfig.cmake
CMake Warning at cmake/modules/FindOrFetchCereal.cmake:4 (find_package):
  By not providing "Findcereal.cmake" in CMAKE_MODULE_PATH this project has
  asked CMake to find a package configuration file provided by "cereal", but
  CMake did not find one.

  Could not find a package configuration file provided by "cereal" with any
  of the following names:

    cerealConfig.cmake
    cereal-config.cmake

  Add the installation prefix of "cereal" to CMAKE_PREFIX_PATH or set
  "cereal_DIR" to a directory containing one of the above files.  If "cereal"
  provides a separate development package or SDK, be sure it has been
  installed.
Call Stack (most recent call first):
  CMakeLists.txt:63 (include)


-- The C compiler identification is GNU 10.3.0
-- Check for working C compiler: /usr/bin/gcc-10
-- Check for working C compiler: /usr/bin/gcc-10 -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Looking for sys/types.h
-- Looking for sys/types.h - found
-- Looking for stdint.h
-- Looking for stdint.h - found
-- Looking for stddef.h
-- Looking for stddef.h - found
-- Check size of long double
-- Check size of long double - done
-- Check size of long long
-- Check size of long long - done
CMake Deprecation Warning at build/_deps/blaspp-src/CMakeLists.txt:73 (message):
  WARNING: `use_cuda=OFF` and `use_hip=` are deprecated; setting `gpu_backend
  = none`


-- Using CMAKE_INSTALL_PREFIX = /usr/local
-- Skipping CUDA search. gpu_backend = none
-- Not building CUDA support in BLAS++
-- Skipping HIP/ROCm search. gpu_backend = none
-- Not building HIP/ROCm support in BLAS++
-- Found OpenMP_C: -fopenmp (found version "4.5") 
-- Found OpenMP_CXX: -fopenmp (found version "4.5") 
-- Found OpenMP: TRUE (found version "4.5")  
-- blaspp_id = 0c63c24
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Failed
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
-- Found Threads: TRUE  
-- Looking for sgemm_
-- Looking for sgemm_ - found
-- Found BLAS: /opt/intel/oneapi/mkl/2021.3.0/lib/intel64/libmkl_intel_lp64.so;/opt/intel/oneapi/mkl/2021.3.0/lib/intel64/libmkl_sequential.so;/opt/intel/oneapi/mkl/2021.3.0/lib/intel64/libmkl_core.so;-lpthread;-lm;-ldl  
-- Checking BLAS library version
   MKL_VERSION=2021.0.3

-- Checking BLAS complex return type
   BLAS (zdotc) returns complex as hidden argument (Intel ifort convention)
-- Checking BLAS float return type
   BLAS (sdot) returns float as float (standard)
-- Checking for CBLAS library 
   Found CBLAS library 
-- Using CMAKE_INSTALL_PREFIX = /usr/local
-- lapackpp_id = 13301a1
-- Check for BLAS++
   BLAS++ already included
-- Looking for cheev_
-- Looking for cheev_ - found
-- A library with LAPACK API found.
-- Checking LAPACK version
   Unknown LAPACK version
-- Checking for XBLAS
   XBLAS not found.
-- Checking for LAPACKE library 
   Found LAPACKE library in BLAS library
-- Looking for mkl_dimatcopy
-- Looking for mkl_dimatcopy - found
-- Could NOT find boost_serialization (missing: boost_serialization_DIR)
-- Performing Test BOOST_COMPILES_AND_RUNS
-- Performing Test BOOST_COMPILES_AND_RUNS - Failed
CMake Warning (dev) in build/_deps/btas-src/external/boost.cmake:
  A logical block opening on the line

    /home/joseph/src/ttg/ttg/build/_deps/btas-src/external/boost.cmake:120 (if)

  closes on the line

    /home/joseph/src/ttg/ttg/build/_deps/btas-src/external/boost.cmake:124 (endif)

  with mis-matching arguments.
Call Stack (most recent call first):
  build/_deps/btas-src/CMakeLists.txt:119 (include)
This warning is for project developers.  Use -Wno-dev to suppress it.

-- Boost found at , but could not compile and/or run test program
CMake Warning at build/_deps/btas-src/external/boost.cmake:122 (message):
  To obtain usable Boost, use your system package manager (HomeBrew, apt,
  etc.) OR download at www.boost.org and compile (unpacking alone is not
  enough)
Call Stack (most recent call first):
  build/_deps/btas-src/CMakeLists.txt:119 (include)


CMake Warning at build/_deps/btas-src/external/boost.cmake:123 (message):
  ** !! due to missing Boost.Serialization the corresponding unit tests will
  be disabled !!
Call Stack (most recent call first):
  build/_deps/btas-src/CMakeLists.txt:119 (include)


-- Found Doxygen: /usr/bin/doxygen (found version "1.8.17") found components: doxygen missing components: dot
-- Found Git: /usr/bin/git (found version "2.25.1") 
-- === BTAS Package/Feature Info ===
-- The following features have been enabled:

 * ASSERT_THROWS, BTAS_ASSERT(x) will throw if x is false, and not be affected by NDEBUG
 * USE_BLAS_LAPACK, Will use BLAS and LAPACK linear algebra distributions via their BLAS++/LAPACK++ interfaces
 * TARGET_MAX_INDEX_RANK=6, default BTAS index type will use stack for rank<=6

-- The following OPTIONAL packages have been found:

 * Eigen3
 * BLAS
 * Threads
 * LAPACK
 * boost_headers (required version == 1.71.0)
 * Boost
 * Doxygen

-- The following REQUIRED packages have been found:

 * Git

-- The following features have been disabled:

 * BUILD_DEPS_FROM_SOURCE, Will build missing dependencies from source

-- The following OPTIONAL packages have not been found:

 * cereal
 * boost_serialization (required version == 1.71.0)

-- Found OpenMP_CXX: -fopenmp  
-- Found OpenMP: TRUE  found components: CXX 
-- Performing Test CXX_EXECUTION_NO_LINK_NEEDED
-- Performing Test CXX_EXECUTION_NO_LINK_NEEDED - Success
-- Found CXXStdExecution: 1  
-- Found HWLOC: /usr/lib/x86_64-linux-gnu/libhwloc.so  
-- Performing Test PARSEC_HAVE_HWLOC_PARENT_MEMBER
-- Performing Test PARSEC_HAVE_HWLOC_PARENT_MEMBER - Success
-- Performing Test PARSEC_HAVE_HWLOC_CACHE_ATTR
-- Performing Test PARSEC_HAVE_HWLOC_CACHE_ATTR - Success
-- Performing Test PARSEC_HAVE_HWLOC_OBJ_PU
-- Performing Test PARSEC_HAVE_HWLOC_OBJ_PU - Success
-- Looking for hwloc_bitmap_free in /usr/lib/x86_64-linux-gnu/libhwloc.so
-- Looking for hwloc_bitmap_free in /usr/lib/x86_64-linux-gnu/libhwloc.so - found
-- Found PAPI: /usr/lib/x86_64-linux-gnu/libpapi.so  
-- PAPI Library found at /usr/include /usr/lib/x86_64-linux-gnu/libpapi.so
-- Found MPI_C: /home/joseph/opt/openmpi-v4.1.x/lib/libmpi.so (found version "3.1") 
-- Found MPI: TRUE (found version "3.1") found components: C 
-- Found PaRSEC: PaRSEC_CONFIG=/home/joseph/opt/parsec-termdet/share/cmake/parsec/PaRSECConfig.cmake
-- PARSEC_FOUND=1
-- The ASM compiler identification is GNU
-- Found assembler: /usr/bin/gcc-10
-- Assertion type: assert
-- Performing Test HAVE_CRAYXE
-- Performing Test HAVE_CRAYXE - Failed
-- Performing Test HAVE_CRAYXT
-- Performing Test HAVE_CRAYXT - Failed
-- Performing Test HAVE_IBMBGP
-- Performing Test HAVE_IBMBGP - Failed
-- Performing Test HAVE_IBMBGQ
-- Performing Test HAVE_IBMBGQ - Failed
-- Performing Test USE_X86_64_ASM
-- Performing Test USE_X86_64_ASM - Success
-- Looking for sys/stat.h
-- Looking for sys/stat.h - found
-- Looking for unistd.h
-- Looking for unistd.h - found
-- Check size of int64_t
-- Check size of int64_t - done
-- Check size of long double
-- Check size of long double - done
-- Check size of long long
-- Check size of long long - done
-- Performing Test SYS_TYPES_H_HAS_PID_T
-- Performing Test SYS_TYPES_H_HAS_PID_T - Success
-- Looking for fork
-- Looking for fork - found
-- Looking for memset
-- Looking for memset - found
-- Looking for posix_memalign
-- Looking for posix_memalign - found
-- Performing Test CHECK_STDLIB_H_HAS_POSIX_MEMALIGN_THROW
-- Performing Test CHECK_STDLIB_H_HAS_POSIX_MEMALIGN_THROW - Success
-- Looking for pow
-- Looking for pow - not found
-- Looking for random
-- Looking for random - found
-- Looking for sleep
-- Looking for sleep - found
-- Looking for strchr
-- Looking for strchr - found
-- Performing Test HAVE_STD_ABS_LONG
-- Performing Test HAVE_STD_ABS_LONG - Success
-- Performing Test THREAD_LOCAL_SUPPORT
-- Performing Test THREAD_LOCAL_SUPPORT - Success
-- Thread local keyword: thread_local
-- Performing Test RESTRICT_SUPPORT
-- Performing Test RESTRICT_SUPPORT - Failed
-- Performing Test RESTRICT_SUPPORT
-- Performing Test RESTRICT_SUPPORT - Success
-- Restrict keyword: __restrict
-- Performing Test HAVE_UNQUALIFIED_STATIC_DECL
-- Performing Test HAVE_UNQUALIFIED_STATIC_DECL - Success
-- Checking for PIE-disabling linker flags
-- Performing Test LINKER_HAS_DISABLEPIE_SUPPORT
-- Performing Test LINKER_HAS_DISABLEPIE_SUPPORT - Success
-- PIE-disabling linker flags: -no-pie
-- Found LATEX: /usr/bin/latex   
-- Found xterm: /usr/bin/xterm
-- Found gdb: /usr/bin/gdb
-- Found MPI_CXX: /home/joseph/opt/openmpi-v4.1.x/lib/libmpi.so (found version "3.1") 
-- Found MPI: TRUE (found version "3.1")  
-- MPI main header: /home/joseph/opt/openmpi-v4.1.x/include/mpi.h
CUDA_TOOLKIT_ROOT_DIR not found or specified
-- Could NOT find CUDA (missing: CUDA_TOOLKIT_ROOT_DIR CUDA_NVCC_EXECUTABLE CUDA_INCLUDE_DIRS CUDA_CUDART_LIBRARY) 
-- === MADNESS Package/Feature Info ===
-- The following features have been enabled:

 * ASSERT_THROWS, BTAS_ASSERT(x) will throw if x is false, and not be affected by NDEBUG
 * USE_BLAS_LAPACK, Will use BLAS and LAPACK linear algebra distributions via their BLAS++/LAPACK++ interfaces
 * TARGET_MAX_INDEX_RANK=6, default BTAS index type will use stack for rank<=6
 * SPINLOCKS, Enables use of spinlocks instead of mutexes (faster unless over subscribing processors)
 * DQ_PREBUF, Enables thread-local buffer for task aggregation to reduce lock contention
 * BSEND_ACKS, Use MPI Send instead of MPI Bsend for huge message acknowledgements
 * FORTRAN_INTEGER_SIZE=4, assumes Fortran integers to be 4 bytes long
 * ASSERTION_TYPE=assert, controls how MADNESS assertions (MADNESS_ASSERT) are handled
 * MPI_THREAD=multiple, controls the level of thread-safety support in MPI

-- The following OPTIONAL packages have been found:

 * Eigen3
 * BLAS
 * LAPACK
 * boost_headers (required version == 1.71.0)
 * Boost
 * OpenMP
 * CXXStdExecution
 * Threads
 * Doxygen
 * LATEX

-- The following REQUIRED packages have been found:

 * HWLOC
 * PAPI
 * MPI
 * Git

-- The following features have been disabled:

 * BUILD_DEPS_FROM_SOURCE, Will build missing dependencies from source
 * Elemental, compiles parallel linear-algebra library Elemental as part of MADNESS
 * GENTENSOR, enables use of MRA Function compression that allows computing in 6 dimensions
 * TASK_PROFILER, supports task-level tracing of program execution
 * WORLD_PROFILE, supports simple profiling of MADworld runtime
 * MEM_STATS, gather fine-grained memory statistics (expensive)
 * MEM_PROFILE, instrumented aggregate memory profiling
 * TENSOR_BOUNDS_CHECKING, Enable checking of bounds in tensors ... slow but useful for debugging
 * TENSOR_INSTANCE_COUNT, Enable counting of allocated tensors for memory leak detection
 * NEVER_SPIN, Disables use of spinlocks (notably for use inside virtual machines)
 * WORLD_GET_DEFAULT_DISABLE, Disables World::get_default()
 * TASK_DEBUG_TRACE, supports debug trace of task engine
 * ASSUMES_ASLR_DISABLED, MADNESS runtime assumes the Address Space Layout Randomization (ASLR) to be disabled

-- The following OPTIONAL packages have not been found:

 * cereal
 * boost_serialization (required version == 1.71.0)
 * CUDA

-- MADNESS_FOUND=1
-- Performing Test BLAS_HAS_LAPACK
-- Performing Test BLAS_HAS_LAPACK - Success
-- Performing Test BLAS_HAS_CBLAS
-- Performing Test BLAS_HAS_CBLAS - Success
-- Performing Test BLAS_HAS_LAPACKE
-- Performing Test BLAS_HAS_LAPACKE - Success
-- Found LAPACKE: /opt/intel/oneapi/mkl/2021.3.0/lib/intel64/libmkl_intel_lp64.so;/opt/intel/oneapi/mkl/2021.3.0/lib/intel64/libmkl_sequential.so;/opt/intel/oneapi/mkl/2021.3.0/lib/intel64/libmkl_core.so;-lpthread;-lm;-ldl  found components: BLAS CBLAS LAPACK LAPACKE 
-- Found LAPACKE and defined the following imported targets:
--   - LAPACKE::LAPACKE:
--       + include:      
--       + library:      /opt/intel/oneapi/mkl/2021.3.0/lib/intel64/libmkl_intel_lp64.so
--       + dependencies: /opt/intel/oneapi/mkl/2021.3.0/lib/intel64/libmkl_sequential.so;/opt/intel/oneapi/mkl/2021.3.0/lib/intel64/libmkl_core.so;-lpthread;-lm;-ldl;;m
--   - LAPACKE::LAPACK:
--       + include:      
--       + library:      /opt/intel/oneapi/mkl/2021.3.0/lib/intel64/libmkl_intel_lp64.so
--       + dependencies: /opt/intel/oneapi/mkl/2021.3.0/lib/intel64/libmkl_sequential.so;/opt/intel/oneapi/mkl/2021.3.0/lib/intel64/libmkl_core.so;-lpthread;-lm;-ldl;;m
--   - LAPACKE::CBLAS:
--       + include:      
--       + library:      /opt/intel/oneapi/mkl/2021.3.0/lib/intel64/libmkl_intel_lp64.so
--       + dependencies: /opt/intel/oneapi/mkl/2021.3.0/lib/intel64/libmkl_sequential.so;/opt/intel/oneapi/mkl/2021.3.0/lib/intel64/libmkl_core.so;-lpthread;-lm;-ldl;;m
--   - LAPACKE::BLAS:
--       + include:      
--       + library:      /opt/intel/oneapi/mkl/2021.3.0/lib/intel64/libmkl_intel_lp64.so
--       + dependencies: /opt/intel/oneapi/mkl/2021.3.0/lib/intel64/libmkl_sequential.so;/opt/intel/oneapi/mkl/2021.3.0/lib/intel64/libmkl_core.so;-lpthread;-lm;-ldl;/opt/intel/oneapi/mkl/2021.3.0/lib/intel64/libmkl_intel_lp64.so;/opt/intel/oneapi/mkl/2021.3.0/lib/intel64/libmkl_sequential.so;/opt/intel/oneapi/mkl/2021.3.0/lib/intel64/libmkl_core.so;-lpthread;-lm;-ldl;m
-- Found Doxygen: /usr/bin/doxygen (found suitable version "1.8.17", minimum required is "1.8.12") found components: doxygen missing components: dot
-- CMAKE_INSTALL_CMAKEDIR=lib/cmake/ttg
-- Configuring done
-- Generating done
CMake Warning:
  Manually-specified variables were not used by the project:

    CMAKE_Fortran_COMPILER


-- Build files have been written to: /home/joseph/src/ttg/ttg/build
```

@devreal what happens if you omit the `BLA_VENDOR` argument? I have seen
this and suspect that this has something to do with BLAS++ being broken
when using standard FindBLAS module rather than its own built-in BLAS
discovery


On Thu, Sep 2, 2021 at 7:31 PM Joseph Schuchart ***@***.***>
wrote:

> Trying to build TTG fails on current master:
>
> [1/426] Building CXX object _deps/lapackpp-build/CMakeFiles/lapackpp.dir/src/bbcsd.cc.o
>
> FAILED: _deps/lapackpp-build/CMakeFiles/lapackpp.dir/src/bbcsd.cc.o
>
> /usr/bin/g++-10  -DLAPACKPP_ID=\"13301a1\" -I_deps/lapackpp-build/include -I_deps/lapackpp-src/include -I_deps/blaspp-build/include -I_deps/blaspp-src/include -g -fPIC   -fdiagnostics-color=always -Wall -fopenmp -std=c++17 -MD -MT _deps/lapackpp-build/CMakeFiles/lapackpp.dir/src/bbcsd.cc.o -MF _deps/lapackpp-build/CMakeFiles/lapackpp.dir/src/bbcsd.cc.o.d -o _deps/lapackpp-build/CMakeFiles/lapackpp.dir/src/bbcsd.cc.o -c _deps/lapackpp-src/src/bbcsd.cc
>
> In file included from _deps/lapackpp-src/include/lapack/fortran.h:10,
>
>                  from _deps/lapackpp-src/src/bbcsd.cc:7:
>
> _deps/lapackpp-src/include/lapack/mangling.h:26:10: error: #error "One of LAPACK_FORTRAN_ADD_, LAPACK_FORTRAN_LOWER, or LAPACK_FORTRAN_UPPER must be defined to set how Fortran functions are name mangled."
>
>    26 |         #error "One of LAPACK_FORTRAN_ADD_, LAPACK_FORTRAN_LOWER, or LAPACK_FORTRAN_UPPER must be defined to set how Fortran functions are name mangled."
>
>       |          ^~~~~
>
> In file included from _deps/lapackpp-src/src/bbcsd.cc:7:
>
> _deps/lapackpp-src/include/lapack/fortran.h:48:23: error: variable or field ‘LAPACK_GLOBAL’ declared void
>
>    48 | #define LAPACK_sgetrf LAPACK_GLOBAL(sgetrf,SGETRF)
>
>       |                       ^~~~~~~~~~~~~
>
> _deps/lapackpp-src/include/lapack/fortran.h:49:6: note: in expansion of macro ‘LAPACK_sgetrf’
>
>    49 | void LAPACK_sgetrf(
>
>       |      ^~~~~~~~~~~~~
>
> _deps/lapackpp-src/include/lapack/fortran.h:48:37: error: ‘sgetrf’ was not declared in this scope
>
>    48 | #define LAPACK_sgetrf LAPACK_GLOBAL(sgetrf,SGETRF)
>
>       |                                     ^~~~~~
>
> _deps/lapackpp-src/include/lapack/fortran.h:49:6: note: in expansion of macro ‘LAPACK_sgetrf’
>
>    49 | void LAPACK_sgetrf(
>
>       |      ^~~~~~~~~~~~~
>
> _deps/lapackpp-src/include/lapack/fortran.h:48:44: error: ‘SGETRF’ was not declared in this scope
>
>    48 | #define LAPACK_sgetrf LAPACK_GLOBAL(sgetrf,SGETRF)
>
>       |                                            ^~~~~~
>
>
> CMake invocation and log:
>
> $ cmake .. -DBUILD_SHARED_LIBS=OFF -DBLA_VENDOR=Intel10_64lp_seq -DBUILD_SHARED_LIBS=OFF -Duse_cuda=OFF -DCMAKE_BUILD_TYPE=Debug -DDPLASMA_ROOT_DIR=$HOME/opt/dplasma-termdet -DPaRSEC_ROOT_DIR=$HOME/opt/parsec-termdet/ -DCMAKE_C_COMPILER=gcc-10 -DCMAKE_CXX_COMPILER=g++-10 -DCMAKE_Fortran_COMPILER=g++-10 -G Ninja
>
> -- The CXX compiler identification is GNU 10.3.0
>
> -- Check for working CXX compiler: /usr/bin/g++-10
>
> -- Check for working CXX compiler: /usr/bin/g++-10 -- works
>
> -- Detecting CXX compiler ABI info
>
> -- Detecting CXX compiler ABI info - done
>
> -- Detecting CXX compile features
>
> -- Detecting CXX compile features - done
>
> -- EIGEN3_FOUND=1
>
> -- Found Boost at /usr/lib/x86_64-linux-gnu/cmake/Boost-1.71.0/BoostConfig.cmake
>
> CMake Warning at cmake/modules/FindOrFetchCereal.cmake:4 (find_package):
>
>   By not providing "Findcereal.cmake" in CMAKE_MODULE_PATH this project has
>
>   asked CMake to find a package configuration file provided by "cereal", but
>
>   CMake did not find one.
>
>
>
>   Could not find a package configuration file provided by "cereal" with any
>
>   of the following names:
>
>
>
>     cerealConfig.cmake
>
>     cereal-config.cmake
>
>
>
>   Add the installation prefix of "cereal" to CMAKE_PREFIX_PATH or set
>
>   "cereal_DIR" to a directory containing one of the above files.  If "cereal"
>
>   provides a separate development package or SDK, be sure it has been
>
>   installed.
>
> Call Stack (most recent call first):
>
>   CMakeLists.txt:63 (include)
>
>
>
>
>
> -- The C compiler identification is GNU 10.3.0
>
> -- Check for working C compiler: /usr/bin/gcc-10
>
> -- Check for working C compiler: /usr/bin/gcc-10 -- works
>
> -- Detecting C compiler ABI info
>
> -- Detecting C compiler ABI info - done
>
> -- Detecting C compile features
>
> -- Detecting C compile features - done
>
> -- Looking for sys/types.h
>
> -- Looking for sys/types.h - found
>
> -- Looking for stdint.h
>
> -- Looking for stdint.h - found
>
> -- Looking for stddef.h
>
> -- Looking for stddef.h - found
>
> -- Check size of long double
>
> -- Check size of long double - done
>
> -- Check size of long long
>
> -- Check size of long long - done
>
> CMake Deprecation Warning at build/_deps/blaspp-src/CMakeLists.txt:73 (message):
>
>   WARNING: `use_cuda=OFF` and `use_hip=` are deprecated; setting `gpu_backend
>
>   = none`
>
>
>
>
>
> -- Using CMAKE_INSTALL_PREFIX = /usr/local
>
> -- Skipping CUDA search. gpu_backend = none
>
> -- Not building CUDA support in BLAS++
>
> -- Skipping HIP/ROCm search. gpu_backend = none
>
> -- Not building HIP/ROCm support in BLAS++
>
> -- Found OpenMP_C: -fopenmp (found version "4.5")
>
> -- Found OpenMP_CXX: -fopenmp (found version "4.5")
>
> -- Found OpenMP: TRUE (found version "4.5")
>
> -- blaspp_id = 0c63c24
>
> -- Looking for pthread.h
>
> -- Looking for pthread.h - found
>
> -- Performing Test CMAKE_HAVE_LIBC_PTHREAD
>
> -- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Failed
>
> -- Looking for pthread_create in pthreads
>
> -- Looking for pthread_create in pthreads - not found
>
> -- Looking for pthread_create in pthread
>
> -- Looking for pthread_create in pthread - found
>
> -- Found Threads: TRUE
>
> -- Looking for sgemm_
>
> -- Looking for sgemm_ - found
>
> -- Found BLAS: /opt/intel/oneapi/mkl/2021.3.0/lib/intel64/libmkl_intel_lp64.so;/opt/intel/oneapi/mkl/2021.3.0/lib/intel64/libmkl_sequential.so;/opt/intel/oneapi/mkl/2021.3.0/lib/intel64/libmkl_core.so;-lpthread;-lm;-ldl
>
> -- Checking BLAS library version
>
>    MKL_VERSION=2021.0.3
>
>
>
> -- Checking BLAS complex return type
>
>    BLAS (zdotc) returns complex as hidden argument (Intel ifort convention)
>
> -- Checking BLAS float return type
>
>    BLAS (sdot) returns float as float (standard)
>
> -- Checking for CBLAS library
>
>    Found CBLAS library
>
> -- Using CMAKE_INSTALL_PREFIX = /usr/local
>
> -- lapackpp_id = 13301a1
>
> -- Check for BLAS++
>
>    BLAS++ already included
>
> -- Looking for cheev_
>
> -- Looking for cheev_ - found
>
> -- A library with LAPACK API found.
>
> -- Checking LAPACK version
>
>    Unknown LAPACK version
>
> -- Checking for XBLAS
>
>    XBLAS not found.
>
> -- Checking for LAPACKE library
>
>    Found LAPACKE library in BLAS library
>
> -- Looking for mkl_dimatcopy
>
> -- Looking for mkl_dimatcopy - found
>
> -- Could NOT find boost_serialization (missing: boost_serialization_DIR)
>
> -- Performing Test BOOST_COMPILES_AND_RUNS
>
> -- Performing Test BOOST_COMPILES_AND_RUNS - Failed
>
> CMake Warning (dev) in build/_deps/btas-src/external/boost.cmake:
>
>   A logical block opening on the line
>
>
>
>     /home/joseph/src/ttg/ttg/build/_deps/btas-src/external/boost.cmake:120 (if)
>
>
>
>   closes on the line
>
>
>
>     /home/joseph/src/ttg/ttg/build/_deps/btas-src/external/boost.cmake:124 (endif)
>
>
>
>   with mis-matching arguments.
>
> Call Stack (most recent call first):
>
>   build/_deps/btas-src/CMakeLists.txt:119 (include)
>
> This warning is for project developers.  Use -Wno-dev to suppress it.
>
>
>
> -- Boost found at , but could not compile and/or run test program
>
> CMake Warning at build/_deps/btas-src/external/boost.cmake:122 (message):
>
>   To obtain usable Boost, use your system package manager (HomeBrew, apt,
>
>   etc.) OR download at www.boost.org and compile (unpacking alone is not
>
>   enough)
>
> Call Stack (most recent call first):
>
>   build/_deps/btas-src/CMakeLists.txt:119 (include)
>
>
>
>
>
> CMake Warning at build/_deps/btas-src/external/boost.cmake:123 (message):
>
>   ** !! due to missing Boost.Serialization the corresponding unit tests will
>
>   be disabled !!
>
> Call Stack (most recent call first):
>
>   build/_deps/btas-src/CMakeLists.txt:119 (include)
>
>
>
>
>
> -- Found Doxygen: /usr/bin/doxygen (found version "1.8.17") found components: doxygen missing components: dot
>
> -- Found Git: /usr/bin/git (found version "2.25.1")
>
> -- === BTAS Package/Feature Info ===
>
> -- The following features have been enabled:
>
>
>
>  * ASSERT_THROWS, BTAS_ASSERT(x) will throw if x is false, and not be affected by NDEBUG
>
>  * USE_BLAS_LAPACK, Will use BLAS and LAPACK linear algebra distributions via their BLAS++/LAPACK++ interfaces
>
>  * TARGET_MAX_INDEX_RANK=6, default BTAS index type will use stack for rank<=6
>
>
>
> -- The following OPTIONAL packages have been found:
>
>
>
>  * Eigen3
>
>  * BLAS
>
>  * Threads
>
>  * LAPACK
>
>  * boost_headers (required version == 1.71.0)
>
>  * Boost
>
>  * Doxygen
>
>
>
> -- The following REQUIRED packages have been found:
>
>
>
>  * Git
>
>
>
> -- The following features have been disabled:
>
>
>
>  * BUILD_DEPS_FROM_SOURCE, Will build missing dependencies from source
>
>
>
> -- The following OPTIONAL packages have not been found:
>
>
>
>  * cereal
>
>  * boost_serialization (required version == 1.71.0)
>
>
>
> -- Found OpenMP_CXX: -fopenmp
>
> -- Found OpenMP: TRUE  found components: CXX
>
> -- Performing Test CXX_EXECUTION_NO_LINK_NEEDED
>
> -- Performing Test CXX_EXECUTION_NO_LINK_NEEDED - Success
>
> -- Found CXXStdExecution: 1
>
> -- Found HWLOC: /usr/lib/x86_64-linux-gnu/libhwloc.so
>
> -- Performing Test PARSEC_HAVE_HWLOC_PARENT_MEMBER
>
> -- Performing Test PARSEC_HAVE_HWLOC_PARENT_MEMBER - Success
>
> -- Performing Test PARSEC_HAVE_HWLOC_CACHE_ATTR
>
> -- Performing Test PARSEC_HAVE_HWLOC_CACHE_ATTR - Success
>
> -- Performing Test PARSEC_HAVE_HWLOC_OBJ_PU
>
> -- Performing Test PARSEC_HAVE_HWLOC_OBJ_PU - Success
>
> -- Looking for hwloc_bitmap_free in /usr/lib/x86_64-linux-gnu/libhwloc.so
>
> -- Looking for hwloc_bitmap_free in /usr/lib/x86_64-linux-gnu/libhwloc.so - found
>
> -- Found PAPI: /usr/lib/x86_64-linux-gnu/libpapi.so
>
> -- PAPI Library found at /usr/include /usr/lib/x86_64-linux-gnu/libpapi.so
>
> -- Found MPI_C: /home/joseph/opt/openmpi-v4.1.x/lib/libmpi.so (found version "3.1")
>
> -- Found MPI: TRUE (found version "3.1") found components: C
>
> -- Found PaRSEC: PaRSEC_CONFIG=/home/joseph/opt/parsec-termdet/share/cmake/parsec/PaRSECConfig.cmake
>
> -- PARSEC_FOUND=1
>
> -- The ASM compiler identification is GNU
>
> -- Found assembler: /usr/bin/gcc-10
>
> -- Assertion type: assert
>
> -- Performing Test HAVE_CRAYXE
>
> -- Performing Test HAVE_CRAYXE - Failed
>
> -- Performing Test HAVE_CRAYXT
>
> -- Performing Test HAVE_CRAYXT - Failed
>
> -- Performing Test HAVE_IBMBGP
>
> -- Performing Test HAVE_IBMBGP - Failed
>
> -- Performing Test HAVE_IBMBGQ
>
> -- Performing Test HAVE_IBMBGQ - Failed
>
> -- Performing Test USE_X86_64_ASM
>
> -- Performing Test USE_X86_64_ASM - Success
>
> -- Looking for sys/stat.h
>
> -- Looking for sys/stat.h - found
>
> -- Looking for unistd.h
>
> -- Looking for unistd.h - found
>
> -- Check size of int64_t
>
> -- Check size of int64_t - done
>
> -- Check size of long double
>
> -- Check size of long double - done
>
> -- Check size of long long
>
> -- Check size of long long - done
>
> -- Performing Test SYS_TYPES_H_HAS_PID_T
>
> -- Performing Test SYS_TYPES_H_HAS_PID_T - Success
>
> -- Looking for fork
>
> -- Looking for fork - found
>
> -- Looking for memset
>
> -- Looking for memset - found
>
> -- Looking for posix_memalign
>
> -- Looking for posix_memalign - found
>
> -- Performing Test CHECK_STDLIB_H_HAS_POSIX_MEMALIGN_THROW
>
> -- Performing Test CHECK_STDLIB_H_HAS_POSIX_MEMALIGN_THROW - Success
>
> -- Looking for pow
>
> -- Looking for pow - not found
>
> -- Looking for random
>
> -- Looking for random - found
>
> -- Looking for sleep
>
> -- Looking for sleep - found
>
> -- Looking for strchr
>
> -- Looking for strchr - found
>
> -- Performing Test HAVE_STD_ABS_LONG
>
> -- Performing Test HAVE_STD_ABS_LONG - Success
>
> -- Performing Test THREAD_LOCAL_SUPPORT
>
> -- Performing Test THREAD_LOCAL_SUPPORT - Success
>
> -- Thread local keyword: thread_local
>
> -- Performing Test RESTRICT_SUPPORT
>
> -- Performing Test RESTRICT_SUPPORT - Failed
>
> -- Performing Test RESTRICT_SUPPORT
>
> -- Performing Test RESTRICT_SUPPORT - Success
>
> -- Restrict keyword: __restrict
>
> -- Performing Test HAVE_UNQUALIFIED_STATIC_DECL
>
> -- Performing Test HAVE_UNQUALIFIED_STATIC_DECL - Success
>
> -- Checking for PIE-disabling linker flags
>
> -- Performing Test LINKER_HAS_DISABLEPIE_SUPPORT
>
> -- Performing Test LINKER_HAS_DISABLEPIE_SUPPORT - Success
>
> -- PIE-disabling linker flags: -no-pie
>
> -- Found LATEX: /usr/bin/latex
>
> -- Found xterm: /usr/bin/xterm
>
> -- Found gdb: /usr/bin/gdb
>
> -- Found MPI_CXX: /home/joseph/opt/openmpi-v4.1.x/lib/libmpi.so (found version "3.1")
>
> -- Found MPI: TRUE (found version "3.1")
>
> -- MPI main header: /home/joseph/opt/openmpi-v4.1.x/include/mpi.h
>
> CUDA_TOOLKIT_ROOT_DIR not found or specified
>
> -- Could NOT find CUDA (missing: CUDA_TOOLKIT_ROOT_DIR CUDA_NVCC_EXECUTABLE CUDA_INCLUDE_DIRS CUDA_CUDART_LIBRARY)
>
> -- === MADNESS Package/Feature Info ===
>
> -- The following features have been enabled:
>
>
>
>  * ASSERT_THROWS, BTAS_ASSERT(x) will throw if x is false, and not be affected by NDEBUG
>
>  * USE_BLAS_LAPACK, Will use BLAS and LAPACK linear algebra distributions via their BLAS++/LAPACK++ interfaces
>
>  * TARGET_MAX_INDEX_RANK=6, default BTAS index type will use stack for rank<=6
>
>  * SPINLOCKS, Enables use of spinlocks instead of mutexes (faster unless over subscribing processors)
>
>  * DQ_PREBUF, Enables thread-local buffer for task aggregation to reduce lock contention
>
>  * BSEND_ACKS, Use MPI Send instead of MPI Bsend for huge message acknowledgements
>
>  * FORTRAN_INTEGER_SIZE=4, assumes Fortran integers to be 4 bytes long
>
>  * ASSERTION_TYPE=assert, controls how MADNESS assertions (MADNESS_ASSERT) are handled
>
>  * MPI_THREAD=multiple, controls the level of thread-safety support in MPI
>
>
>
> -- The following OPTIONAL packages have been found:
>
>
>
>  * Eigen3
>
>  * BLAS
>
>  * LAPACK
>
>  * boost_headers (required version == 1.71.0)
>
>  * Boost
>
>  * OpenMP
>
>  * CXXStdExecution
>
>  * Threads
>
>  * Doxygen
>
>  * LATEX
>
>
>
> -- The following REQUIRED packages have been found:
>
>
>
>  * HWLOC
>
>  * PAPI
>
>  * MPI
>
>  * Git
>
>
>
> -- The following features have been disabled:
>
>
>
>  * BUILD_DEPS_FROM_SOURCE, Will build missing dependencies from source
>
>  * Elemental, compiles parallel linear-algebra library Elemental as part of MADNESS
>
>  * GENTENSOR, enables use of MRA Function compression that allows computing in 6 dimensions
>
>  * TASK_PROFILER, supports task-level tracing of program execution
>
>  * WORLD_PROFILE, supports simple profiling of MADworld runtime
>
>  * MEM_STATS, gather fine-grained memory statistics (expensive)
>
>  * MEM_PROFILE, instrumented aggregate memory profiling
>
>  * TENSOR_BOUNDS_CHECKING, Enable checking of bounds in tensors ... slow but useful for debugging
>
>  * TENSOR_INSTANCE_COUNT, Enable counting of allocated tensors for memory leak detection
>
>  * NEVER_SPIN, Disables use of spinlocks (notably for use inside virtual machines)
>
>  * WORLD_GET_DEFAULT_DISABLE, Disables World::get_default()
>
>  * TASK_DEBUG_TRACE, supports debug trace of task engine
>
>  * ASSUMES_ASLR_DISABLED, MADNESS runtime assumes the Address Space Layout Randomization (ASLR) to be disabled
>
>
>
> -- The following OPTIONAL packages have not been found:
>
>
>
>  * cereal
>
>  * boost_serialization (required version == 1.71.0)
>
>  * CUDA
>
>
>
> -- MADNESS_FOUND=1
>
> -- Performing Test BLAS_HAS_LAPACK
>
> -- Performing Test BLAS_HAS_LAPACK - Success
>
> -- Performing Test BLAS_HAS_CBLAS
>
> -- Performing Test BLAS_HAS_CBLAS - Success
>
> -- Performing Test BLAS_HAS_LAPACKE
>
> -- Performing Test BLAS_HAS_LAPACKE - Success
>
> -- Found LAPACKE: /opt/intel/oneapi/mkl/2021.3.0/lib/intel64/libmkl_intel_lp64.so;/opt/intel/oneapi/mkl/2021.3.0/lib/intel64/libmkl_sequential.so;/opt/intel/oneapi/mkl/2021.3.0/lib/intel64/libmkl_core.so;-lpthread;-lm;-ldl  found components: BLAS CBLAS LAPACK LAPACKE
>
> -- Found LAPACKE and defined the following imported targets:
>
> --   - LAPACKE::LAPACKE:
>
> --       + include:
>
> --       + library:      /opt/intel/oneapi/mkl/2021.3.0/lib/intel64/libmkl_intel_lp64.so
>
> --       + dependencies: /opt/intel/oneapi/mkl/2021.3.0/lib/intel64/libmkl_sequential.so;/opt/intel/oneapi/mkl/2021.3.0/lib/intel64/libmkl_core.so;-lpthread;-lm;-ldl;;m
>
> --   - LAPACKE::LAPACK:
>
> --       + include:
>
> --       + library:      /opt/intel/oneapi/mkl/2021.3.0/lib/intel64/libmkl_intel_lp64.so
>
> --       + dependencies: /opt/intel/oneapi/mkl/2021.3.0/lib/intel64/libmkl_sequential.so;/opt/intel/oneapi/mkl/2021.3.0/lib/intel64/libmkl_core.so;-lpthread;-lm;-ldl;;m
>
> --   - LAPACKE::CBLAS:
>
> --       + include:
>
> --       + library:      /opt/intel/oneapi/mkl/2021.3.0/lib/intel64/libmkl_intel_lp64.so
>
> --       + dependencies: /opt/intel/oneapi/mkl/2021.3.0/lib/intel64/libmkl_sequential.so;/opt/intel/oneapi/mkl/2021.3.0/lib/intel64/libmkl_core.so;-lpthread;-lm;-ldl;;m
>
> --   - LAPACKE::BLAS:
>
> --       + include:
>
> --       + library:      /opt/intel/oneapi/mkl/2021.3.0/lib/intel64/libmkl_intel_lp64.so
>
> --       + dependencies: /opt/intel/oneapi/mkl/2021.3.0/lib/intel64/libmkl_sequential.so;/opt/intel/oneapi/mkl/2021.3.0/lib/intel64/libmkl_core.so;-lpthread;-lm;-ldl;/opt/intel/oneapi/mkl/2021.3.0/lib/intel64/libmkl_intel_lp64.so;/opt/intel/oneapi/mkl/2021.3.0/lib/intel64/libmkl_sequential.so;/opt/intel/oneapi/mkl/2021.3.0/lib/intel64/libmkl_core.so;-lpthread;-lm;-ldl;m
>
> -- Found Doxygen: /usr/bin/doxygen (found suitable version "1.8.17", minimum required is "1.8.12") found components: doxygen missing components: dot
>
> -- CMAKE_INSTALL_CMAKEDIR=lib/cmake/ttg
>
> -- Configuring done
>
> -- Generating done
>
> CMake Warning:
>
>   Manually-specified variables were not used by the project:
>
>
>
>     CMAKE_Fortran_COMPILER
>
>
>
>
>
> -- Build files have been written to: /home/joseph/src/ttg/ttg/build
>
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/TESSEorg/ttg/issues/141>, or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAQXIZ22MCKJQRW2BXZKU2LUAACOLANCNFSM5DKF52NQ>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
>


-- 
web: valeyev.net

I can reproduce with ttg master, and I have -DBLA_VENDOR=Intel10_64lp_seq
in my cmake line.

Cmake shows:
-- Found BLAS:
/usr/lib/x86_64-linux-gnu/libmkl_intel_lp64.so;/usr/lib/x86_64-linux-gnu/libmkl_sequential.so;/usr/lib/x86_64-linux-gnu/libmkl_core.so;-lpthread;-lm;-ldl

-- Checking BLAS library version
-- Checking BLAS complex return type
[34m   BLAS (zdotc) returns complex as hidden argument (Intel ifort
convention) [39m
-- Checking BLAS float return type
[34m   BLAS (sdot) returns float as float (standard) [39m
-- Checking for CBLAS library
-- Checking for CBLAS library -lcblas
[31m   CBLAS library not found. Tester cannot be built. [39m
-- Using CMAKE_INSTALL_PREFIX = /usr/local
-- lapackpp_id = 13301a1
-- Check for BLAS++
   BLAS++ already included



On Thu, Sep 2, 2021 at 9:16 PM Eduard Valeyev ***@***.***>
wrote:

> @devreal what happens if you omit the `BLA_VENDOR` argument? I have seen
> this and suspect that this has something to do with BLAS++ being broken
> when using standard FindBLAS module rather than its own built-in BLAS
> discovery
>
>
> On Thu, Sep 2, 2021 at 7:31 PM Joseph Schuchart ***@***.***>
> wrote:
>
> > Trying to build TTG fails on current master:
> >
> > [1/426] Building CXX object
> _deps/lapackpp-build/CMakeFiles/lapackpp.dir/src/bbcsd.cc.o
> >
> > FAILED: _deps/lapackpp-build/CMakeFiles/lapackpp.dir/src/bbcsd.cc.o
> >
> > /usr/bin/g++-10 -DLAPACKPP_ID=\"13301a1\" -I_deps/lapackpp-build/include
> -I_deps/lapackpp-src/include -I_deps/blaspp-build/include
> -I_deps/blaspp-src/include -g -fPIC -fdiagnostics-color=always -Wall
> -fopenmp -std=c++17 -MD -MT
> _deps/lapackpp-build/CMakeFiles/lapackpp.dir/src/bbcsd.cc.o -MF
> _deps/lapackpp-build/CMakeFiles/lapackpp.dir/src/bbcsd.cc.o.d -o
> _deps/lapackpp-build/CMakeFiles/lapackpp.dir/src/bbcsd.cc.o -c
> _deps/lapackpp-src/src/bbcsd.cc
> >
> > In file included from _deps/lapackpp-src/include/lapack/fortran.h:10,
> >
> > from _deps/lapackpp-src/src/bbcsd.cc:7:
> >
> > _deps/lapackpp-src/include/lapack/mangling.h:26:10: error: #error "One
> of LAPACK_FORTRAN_ADD_, LAPACK_FORTRAN_LOWER, or LAPACK_FORTRAN_UPPER must
> be defined to set how Fortran functions are name mangled."
> >
> > 26 | #error "One of LAPACK_FORTRAN_ADD_, LAPACK_FORTRAN_LOWER, or
> LAPACK_FORTRAN_UPPER must be defined to set how Fortran functions are name
> mangled."
> >
> > | ^~~~~
> >
> > In file included from _deps/lapackpp-src/src/bbcsd.cc:7:
> >
> > _deps/lapackpp-src/include/lapack/fortran.h:48:23: error: variable or
> field ‘LAPACK_GLOBAL’ declared void
> >
> > 48 | #define LAPACK_sgetrf LAPACK_GLOBAL(sgetrf,SGETRF)
> >
> > | ^~~~~~~~~~~~~
> >
> > _deps/lapackpp-src/include/lapack/fortran.h:49:6: note: in expansion of
> macro ‘LAPACK_sgetrf’
> >
> > 49 | void LAPACK_sgetrf(
> >
> > | ^~~~~~~~~~~~~
> >
> > _deps/lapackpp-src/include/lapack/fortran.h:48:37: error: ‘sgetrf’ was
> not declared in this scope
> >
> > 48 | #define LAPACK_sgetrf LAPACK_GLOBAL(sgetrf,SGETRF)
> >
> > | ^~~~~~
> >
> > _deps/lapackpp-src/include/lapack/fortran.h:49:6: note: in expansion of
> macro ‘LAPACK_sgetrf’
> >
> > 49 | void LAPACK_sgetrf(
> >
> > | ^~~~~~~~~~~~~
> >
> > _deps/lapackpp-src/include/lapack/fortran.h:48:44: error: ‘SGETRF’ was
> not declared in this scope
> >
> > 48 | #define LAPACK_sgetrf LAPACK_GLOBAL(sgetrf,SGETRF)
> >
> > | ^~~~~~
> >
> >
> > CMake invocation and log:
> >
> > $ cmake .. -DBUILD_SHARED_LIBS=OFF -DBLA_VENDOR=Intel10_64lp_seq
> -DBUILD_SHARED_LIBS=OFF -Duse_cuda=OFF -DCMAKE_BUILD_TYPE=Debug
> -DDPLASMA_ROOT_DIR=$HOME/opt/dplasma-termdet
> -DPaRSEC_ROOT_DIR=$HOME/opt/parsec-termdet/ -DCMAKE_C_COMPILER=gcc-10
> -DCMAKE_CXX_COMPILER=g++-10 -DCMAKE_Fortran_COMPILER=g++-10 -G Ninja
> >
> > -- The CXX compiler identification is GNU 10.3.0
> >
> > -- Check for working CXX compiler: /usr/bin/g++-10
> >
> > -- Check for working CXX compiler: /usr/bin/g++-10 -- works
> >
> > -- Detecting CXX compiler ABI info
> >
> > -- Detecting CXX compiler ABI info - done
> >
> > -- Detecting CXX compile features
> >
> > -- Detecting CXX compile features - done
> >
> > -- EIGEN3_FOUND=1
> >
> > -- Found Boost at
> /usr/lib/x86_64-linux-gnu/cmake/Boost-1.71.0/BoostConfig.cmake
> >
> > CMake Warning at cmake/modules/FindOrFetchCereal.cmake:4 (find_package):
> >
> > By not providing "Findcereal.cmake" in CMAKE_MODULE_PATH this project has
> >
> > asked CMake to find a package configuration file provided by "cereal",
> but
> >
> > CMake did not find one.
> >
> >
> >
> > Could not find a package configuration file provided by "cereal" with any
> >
> > of the following names:
> >
> >
> >
> > cerealConfig.cmake
> >
> > cereal-config.cmake
> >
> >
> >
> > Add the installation prefix of "cereal" to CMAKE_PREFIX_PATH or set
> >
> > "cereal_DIR" to a directory containing one of the above files. If
> "cereal"
> >
> > provides a separate development package or SDK, be sure it has been
> >
> > installed.
> >
> > Call Stack (most recent call first):
> >
> > CMakeLists.txt:63 (include)
> >
> >
> >
> >
> >
> > -- The C compiler identification is GNU 10.3.0
> >
> > -- Check for working C compiler: /usr/bin/gcc-10
> >
> > -- Check for working C compiler: /usr/bin/gcc-10 -- works
> >
> > -- Detecting C compiler ABI info
> >
> > -- Detecting C compiler ABI info - done
> >
> > -- Detecting C compile features
> >
> > -- Detecting C compile features - done
> >
> > -- Looking for sys/types.h
> >
> > -- Looking for sys/types.h - found
> >
> > -- Looking for stdint.h
> >
> > -- Looking for stdint.h - found
> >
> > -- Looking for stddef.h
> >
> > -- Looking for stddef.h - found
> >
> > -- Check size of long double
> >
> > -- Check size of long double - done
> >
> > -- Check size of long long
> >
> > -- Check size of long long - done
> >
> > CMake Deprecation Warning at build/_deps/blaspp-src/CMakeLists.txt:73
> (message):
> >
> > WARNING: `use_cuda=OFF` and `use_hip=` are deprecated; setting
> `gpu_backend
> >
> > = none`
> >
> >
> >
> >
> >
> > -- Using CMAKE_INSTALL_PREFIX = /usr/local
> >
> > -- Skipping CUDA search. gpu_backend = none
> >
> > -- Not building CUDA support in BLAS++
> >
> > -- Skipping HIP/ROCm search. gpu_backend = none
> >
> > -- Not building HIP/ROCm support in BLAS++
> >
> > -- Found OpenMP_C: -fopenmp (found version "4.5")
> >
> > -- Found OpenMP_CXX: -fopenmp (found version "4.5")
> >
> > -- Found OpenMP: TRUE (found version "4.5")
> >
> > -- blaspp_id = 0c63c24
> >
> > -- Looking for pthread.h
> >
> > -- Looking for pthread.h - found
> >
> > -- Performing Test CMAKE_HAVE_LIBC_PTHREAD
> >
> > -- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Failed
> >
> > -- Looking for pthread_create in pthreads
> >
> > -- Looking for pthread_create in pthreads - not found
> >
> > -- Looking for pthread_create in pthread
> >
> > -- Looking for pthread_create in pthread - found
> >
> > -- Found Threads: TRUE
> >
> > -- Looking for sgemm_
> >
> > -- Looking for sgemm_ - found
> >
> > -- Found BLAS:
> /opt/intel/oneapi/mkl/2021.3.0/lib/intel64/libmkl_intel_lp64.so;/opt/intel/oneapi/mkl/2021.3.0/lib/intel64/libmkl_sequential.so;/opt/intel/oneapi/mkl/2021.3.0/lib/intel64/libmkl_core.so;-lpthread;-lm;-ldl
> >
> > -- Checking BLAS library version
> >
> > MKL_VERSION=2021.0.3
> >
> >
> >
> > -- Checking BLAS complex return type
> >
> > BLAS (zdotc) returns complex as hidden argument (Intel ifort convention)
> >
> > -- Checking BLAS float return type
> >
> > BLAS (sdot) returns float as float (standard)
> >
> > -- Checking for CBLAS library
> >
> > Found CBLAS library
> >
> > -- Using CMAKE_INSTALL_PREFIX = /usr/local
> >
> > -- lapackpp_id = 13301a1
> >
> > -- Check for BLAS++
> >
> > BLAS++ already included
> >
> > -- Looking for cheev_
> >
> > -- Looking for cheev_ - found
> >
> > -- A library with LAPACK API found.
> >
> > -- Checking LAPACK version
> >
> > Unknown LAPACK version
> >
> > -- Checking for XBLAS
> >
> > XBLAS not found.
> >
> > -- Checking for LAPACKE library
> >
> > Found LAPACKE library in BLAS library
> >
> > -- Looking for mkl_dimatcopy
> >
> > -- Looking for mkl_dimatcopy - found
> >
> > -- Could NOT find boost_serialization (missing: boost_serialization_DIR)
> >
> > -- Performing Test BOOST_COMPILES_AND_RUNS
> >
> > -- Performing Test BOOST_COMPILES_AND_RUNS - Failed
> >
> > CMake Warning (dev) in build/_deps/btas-src/external/boost.cmake:
> >
> > A logical block opening on the line
> >
> >
> >
> > /home/joseph/src/ttg/ttg/build/_deps/btas-src/external/boost.cmake:120
> (if)
> >
> >
> >
> > closes on the line
> >
> >
> >
> > /home/joseph/src/ttg/ttg/build/_deps/btas-src/external/boost.cmake:124
> (endif)
> >
> >
> >
> > with mis-matching arguments.
> >
> > Call Stack (most recent call first):
> >
> > build/_deps/btas-src/CMakeLists.txt:119 (include)
> >
> > This warning is for project developers. Use -Wno-dev to suppress it.
> >
> >
> >
> > -- Boost found at , but could not compile and/or run test program
> >
> > CMake Warning at build/_deps/btas-src/external/boost.cmake:122 (message):
> >
> > To obtain usable Boost, use your system package manager (HomeBrew, apt,
> >
> > etc.) OR download at www.boost.org and compile (unpacking alone is not
> >
> > enough)
> >
> > Call Stack (most recent call first):
> >
> > build/_deps/btas-src/CMakeLists.txt:119 (include)
> >
> >
> >
> >
> >
> > CMake Warning at build/_deps/btas-src/external/boost.cmake:123 (message):
> >
> > ** !! due to missing Boost.Serialization the corresponding unit tests
> will
> >
> > be disabled !!
> >
> > Call Stack (most recent call first):
> >
> > build/_deps/btas-src/CMakeLists.txt:119 (include)
> >
> >
> >
> >
> >
> > -- Found Doxygen: /usr/bin/doxygen (found version "1.8.17") found
> components: doxygen missing components: dot
> >
> > -- Found Git: /usr/bin/git (found version "2.25.1")
> >
> > -- === BTAS Package/Feature Info ===
> >
> > -- The following features have been enabled:
> >
> >
> >
> > * ASSERT_THROWS, BTAS_ASSERT(x) will throw if x is false, and not be
> affected by NDEBUG
> >
> > * USE_BLAS_LAPACK, Will use BLAS and LAPACK linear algebra distributions
> via their BLAS++/LAPACK++ interfaces
> >
> > * TARGET_MAX_INDEX_RANK=6, default BTAS index type will use stack for
> rank<=6
> >
> >
> >
> > -- The following OPTIONAL packages have been found:
> >
> >
> >
> > * Eigen3
> >
> > * BLAS
> >
> > * Threads
> >
> > * LAPACK
> >
> > * boost_headers (required version == 1.71.0)
> >
> > * Boost
> >
> > * Doxygen
> >
> >
> >
> > -- The following REQUIRED packages have been found:
> >
> >
> >
> > * Git
> >
> >
> >
> > -- The following features have been disabled:
> >
> >
> >
> > * BUILD_DEPS_FROM_SOURCE, Will build missing dependencies from source
> >
> >
> >
> > -- The following OPTIONAL packages have not been found:
> >
> >
> >
> > * cereal
> >
> > * boost_serialization (required version == 1.71.0)
> >
> >
> >
> > -- Found OpenMP_CXX: -fopenmp
> >
> > -- Found OpenMP: TRUE found components: CXX
> >
> > -- Performing Test CXX_EXECUTION_NO_LINK_NEEDED
> >
> > -- Performing Test CXX_EXECUTION_NO_LINK_NEEDED - Success
> >
> > -- Found CXXStdExecution: 1
> >
> > -- Found HWLOC: /usr/lib/x86_64-linux-gnu/libhwloc.so
> >
> > -- Performing Test PARSEC_HAVE_HWLOC_PARENT_MEMBER
> >
> > -- Performing Test PARSEC_HAVE_HWLOC_PARENT_MEMBER - Success
> >
> > -- Performing Test PARSEC_HAVE_HWLOC_CACHE_ATTR
> >
> > -- Performing Test PARSEC_HAVE_HWLOC_CACHE_ATTR - Success
> >
> > -- Performing Test PARSEC_HAVE_HWLOC_OBJ_PU
> >
> > -- Performing Test PARSEC_HAVE_HWLOC_OBJ_PU - Success
> >
> > -- Looking for hwloc_bitmap_free in /usr/lib/x86_64-linux-gnu/libhwloc.so
> >
> > -- Looking for hwloc_bitmap_free in
> /usr/lib/x86_64-linux-gnu/libhwloc.so - found
> >
> > -- Found PAPI: /usr/lib/x86_64-linux-gnu/libpapi.so
> >
> > -- PAPI Library found at /usr/include
> /usr/lib/x86_64-linux-gnu/libpapi.so
> >
> > -- Found MPI_C: /home/joseph/opt/openmpi-v4.1.x/lib/libmpi.so (found
> version "3.1")
> >
> > -- Found MPI: TRUE (found version "3.1") found components: C
> >
> > -- Found PaRSEC:
> PaRSEC_CONFIG=/home/joseph/opt/parsec-termdet/share/cmake/parsec/PaRSECConfig.cmake
> >
> > -- PARSEC_FOUND=1
> >
> > -- The ASM compiler identification is GNU
> >
> > -- Found assembler: /usr/bin/gcc-10
> >
> > -- Assertion type: assert
> >
> > -- Performing Test HAVE_CRAYXE
> >
> > -- Performing Test HAVE_CRAYXE - Failed
> >
> > -- Performing Test HAVE_CRAYXT
> >
> > -- Performing Test HAVE_CRAYXT - Failed
> >
> > -- Performing Test HAVE_IBMBGP
> >
> > -- Performing Test HAVE_IBMBGP - Failed
> >
> > -- Performing Test HAVE_IBMBGQ
> >
> > -- Performing Test HAVE_IBMBGQ - Failed
> >
> > -- Performing Test USE_X86_64_ASM
> >
> > -- Performing Test USE_X86_64_ASM - Success
> >
> > -- Looking for sys/stat.h
> >
> > -- Looking for sys/stat.h - found
> >
> > -- Looking for unistd.h
> >
> > -- Looking for unistd.h - found
> >
> > -- Check size of int64_t
> >
> > -- Check size of int64_t - done
> >
> > -- Check size of long double
> >
> > -- Check size of long double - done
> >
> > -- Check size of long long
> >
> > -- Check size of long long - done
> >
> > -- Performing Test SYS_TYPES_H_HAS_PID_T
> >
> > -- Performing Test SYS_TYPES_H_HAS_PID_T - Success
> >
> > -- Looking for fork
> >
> > -- Looking for fork - found
> >
> > -- Looking for memset
> >
> > -- Looking for memset - found
> >
> > -- Looking for posix_memalign
> >
> > -- Looking for posix_memalign - found
> >
> > -- Performing Test CHECK_STDLIB_H_HAS_POSIX_MEMALIGN_THROW
> >
> > -- Performing Test CHECK_STDLIB_H_HAS_POSIX_MEMALIGN_THROW - Success
> >
> > -- Looking for pow
> >
> > -- Looking for pow - not found
> >
> > -- Looking for random
> >
> > -- Looking for random - found
> >
> > -- Looking for sleep
> >
> > -- Looking for sleep - found
> >
> > -- Looking for strchr
> >
> > -- Looking for strchr - found
> >
> > -- Performing Test HAVE_STD_ABS_LONG
> >
> > -- Performing Test HAVE_STD_ABS_LONG - Success
> >
> > -- Performing Test THREAD_LOCAL_SUPPORT
> >
> > -- Performing Test THREAD_LOCAL_SUPPORT - Success
> >
> > -- Thread local keyword: thread_local
> >
> > -- Performing Test RESTRICT_SUPPORT
> >
> > -- Performing Test RESTRICT_SUPPORT - Failed
> >
> > -- Performing Test RESTRICT_SUPPORT
> >
> > -- Performing Test RESTRICT_SUPPORT - Success
> >
> > -- Restrict keyword: __restrict
> >
> > -- Performing Test HAVE_UNQUALIFIED_STATIC_DECL
> >
> > -- Performing Test HAVE_UNQUALIFIED_STATIC_DECL - Success
> >
> > -- Checking for PIE-disabling linker flags
> >
> > -- Performing Test LINKER_HAS_DISABLEPIE_SUPPORT
> >
> > -- Performing Test LINKER_HAS_DISABLEPIE_SUPPORT - Success
> >
> > -- PIE-disabling linker flags: -no-pie
> >
> > -- Found LATEX: /usr/bin/latex
> >
> > -- Found xterm: /usr/bin/xterm
> >
> > -- Found gdb: /usr/bin/gdb
> >
> > -- Found MPI_CXX: /home/joseph/opt/openmpi-v4.1.x/lib/libmpi.so (found
> version "3.1")
> >
> > -- Found MPI: TRUE (found version "3.1")
> >
> > -- MPI main header: /home/joseph/opt/openmpi-v4.1.x/include/mpi.h
> >
> > CUDA_TOOLKIT_ROOT_DIR not found or specified
> >
> > -- Could NOT find CUDA (missing: CUDA_TOOLKIT_ROOT_DIR
> CUDA_NVCC_EXECUTABLE CUDA_INCLUDE_DIRS CUDA_CUDART_LIBRARY)
> >
> > -- === MADNESS Package/Feature Info ===
> >
> > -- The following features have been enabled:
> >
> >
> >
> > * ASSERT_THROWS, BTAS_ASSERT(x) will throw if x is false, and not be
> affected by NDEBUG
> >
> > * USE_BLAS_LAPACK, Will use BLAS and LAPACK linear algebra distributions
> via their BLAS++/LAPACK++ interfaces
> >
> > * TARGET_MAX_INDEX_RANK=6, default BTAS index type will use stack for
> rank<=6
> >
> > * SPINLOCKS, Enables use of spinlocks instead of mutexes (faster unless
> over subscribing processors)
> >
> > * DQ_PREBUF, Enables thread-local buffer for task aggregation to reduce
> lock contention
> >
> > * BSEND_ACKS, Use MPI Send instead of MPI Bsend for huge message
> acknowledgements
> >
> > * FORTRAN_INTEGER_SIZE=4, assumes Fortran integers to be 4 bytes long
> >
> > * ASSERTION_TYPE=assert, controls how MADNESS assertions
> (MADNESS_ASSERT) are handled
> >
> > * MPI_THREAD=multiple, controls the level of thread-safety support in MPI
> >
> >
> >
> > -- The following OPTIONAL packages have been found:
> >
> >
> >
> > * Eigen3
> >
> > * BLAS
> >
> > * LAPACK
> >
> > * boost_headers (required version == 1.71.0)
> >
> > * Boost
> >
> > * OpenMP
> >
> > * CXXStdExecution
> >
> > * Threads
> >
> > * Doxygen
> >
> > * LATEX
> >
> >
> >
> > -- The following REQUIRED packages have been found:
> >
> >
> >
> > * HWLOC
> >
> > * PAPI
> >
> > * MPI
> >
> > * Git
> >
> >
> >
> > -- The following features have been disabled:
> >
> >
> >
> > * BUILD_DEPS_FROM_SOURCE, Will build missing dependencies from source
> >
> > * Elemental, compiles parallel linear-algebra library Elemental as part
> of MADNESS
> >
> > * GENTENSOR, enables use of MRA Function compression that allows
> computing in 6 dimensions
> >
> > * TASK_PROFILER, supports task-level tracing of program execution
> >
> > * WORLD_PROFILE, supports simple profiling of MADworld runtime
> >
> > * MEM_STATS, gather fine-grained memory statistics (expensive)
> >
> > * MEM_PROFILE, instrumented aggregate memory profiling
> >
> > * TENSOR_BOUNDS_CHECKING, Enable checking of bounds in tensors ... slow
> but useful for debugging
> >
> > * TENSOR_INSTANCE_COUNT, Enable counting of allocated tensors for memory
> leak detection
> >
> > * NEVER_SPIN, Disables use of spinlocks (notably for use inside virtual
> machines)
> >
> > * WORLD_GET_DEFAULT_DISABLE, Disables World::get_default()
> >
> > * TASK_DEBUG_TRACE, supports debug trace of task engine
> >
> > * ASSUMES_ASLR_DISABLED, MADNESS runtime assumes the Address Space
> Layout Randomization (ASLR) to be disabled
> >
> >
> >
> > -- The following OPTIONAL packages have not been found:
> >
> >
> >
> > * cereal
> >
> > * boost_serialization (required version == 1.71.0)
> >
> > * CUDA
> >
> >
> >
> > -- MADNESS_FOUND=1
> >
> > -- Performing Test BLAS_HAS_LAPACK
> >
> > -- Performing Test BLAS_HAS_LAPACK - Success
> >
> > -- Performing Test BLAS_HAS_CBLAS
> >
> > -- Performing Test BLAS_HAS_CBLAS - Success
> >
> > -- Performing Test BLAS_HAS_LAPACKE
> >
> > -- Performing Test BLAS_HAS_LAPACKE - Success
> >
> > -- Found LAPACKE:
> /opt/intel/oneapi/mkl/2021.3.0/lib/intel64/libmkl_intel_lp64.so;/opt/intel/oneapi/mkl/2021.3.0/lib/intel64/libmkl_sequential.so;/opt/intel/oneapi/mkl/2021.3.0/lib/intel64/libmkl_core.so;-lpthread;-lm;-ldl
> found components: BLAS CBLAS LAPACK LAPACKE
> >
> > -- Found LAPACKE and defined the following imported targets:
> >
> > -- - LAPACKE::LAPACKE:
> >
> > -- + include:
> >
> > -- + library:
> /opt/intel/oneapi/mkl/2021.3.0/lib/intel64/libmkl_intel_lp64.so
> >
> > -- + dependencies:
> /opt/intel/oneapi/mkl/2021.3.0/lib/intel64/libmkl_sequential.so;/opt/intel/oneapi/mkl/2021.3.0/lib/intel64/libmkl_core.so;-lpthread;-lm;-ldl;;m
> >
> > -- - LAPACKE::LAPACK:
> >
> > -- + include:
> >
> > -- + library:
> /opt/intel/oneapi/mkl/2021.3.0/lib/intel64/libmkl_intel_lp64.so
> >
> > -- + dependencies:
> /opt/intel/oneapi/mkl/2021.3.0/lib/intel64/libmkl_sequential.so;/opt/intel/oneapi/mkl/2021.3.0/lib/intel64/libmkl_core.so;-lpthread;-lm;-ldl;;m
> >
> > -- - LAPACKE::CBLAS:
> >
> > -- + include:
> >
> > -- + library:
> /opt/intel/oneapi/mkl/2021.3.0/lib/intel64/libmkl_intel_lp64.so
> >
> > -- + dependencies:
> /opt/intel/oneapi/mkl/2021.3.0/lib/intel64/libmkl_sequential.so;/opt/intel/oneapi/mkl/2021.3.0/lib/intel64/libmkl_core.so;-lpthread;-lm;-ldl;;m
> >
> > -- - LAPACKE::BLAS:
> >
> > -- + include:
> >
> > -- + library:
> /opt/intel/oneapi/mkl/2021.3.0/lib/intel64/libmkl_intel_lp64.so
> >
> > -- + dependencies:
> /opt/intel/oneapi/mkl/2021.3.0/lib/intel64/libmkl_sequential.so;/opt/intel/oneapi/mkl/2021.3.0/lib/intel64/libmkl_core.so;-lpthread;-lm;-ldl;/opt/intel/oneapi/mkl/2021.3.0/lib/intel64/libmkl_intel_lp64.so;/opt/intel/oneapi/mkl/2021.3.0/lib/intel64/libmkl_sequential.so;/opt/intel/oneapi/mkl/2021.3.0/lib/intel64/libmkl_core.so;-lpthread;-lm;-ldl;m
> >
> > -- Found Doxygen: /usr/bin/doxygen (found suitable version "1.8.17",
> minimum required is "1.8.12") found components: doxygen missing components:
> dot
> >
> > -- CMAKE_INSTALL_CMAKEDIR=lib/cmake/ttg
> >
> > -- Configuring done
> >
> > -- Generating done
> >
> > CMake Warning:
> >
> > Manually-specified variables were not used by the project:
> >
> >
> >
> > CMAKE_Fortran_COMPILER
> >
> >
> >
> >
> >
> > -- Build files have been written to: /home/joseph/src/ttg/ttg/build
> >
> >
> > —
> > You are receiving this because you are subscribed to this thread.
> > Reply to this email directly, view it on GitHub
> > <https://github.com/TESSEorg/ttg/issues/141>, or unsubscribe
> > <
> https://github.com/notifications/unsubscribe-auth/AAQXIZ22MCKJQRW2BXZKU2LUAACOLANCNFSM5DKF52NQ
> >
> > .
> > Triage notifications on the go with GitHub Mobile for iOS
> > <
> https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675
> >
> > or Android
> > <
> https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub
> >.
> >
> >
>
>
> --
> web: valeyev.net
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/TESSEorg/ttg/issues/141#issuecomment-912176713>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABFEZNSYVS7DMEJN3PQCTHTUAAOWFANCNFSM5DKF52NQ>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
>

Mhh, if I don't specify `-DBLA_VENDOR` the detection of DPLASMA fails:

```
-- MADNESS_FOUND=1
-- Found BLAS: -lmkl_gf_lp64;-lmkl_gnu_thread;-lmkl_core  
-- Performing Test BLAS_HAS_LAPACK
-- Performing Test BLAS_HAS_LAPACK - Failed
-- Performing Test BLAS_HAS_CBLAS
-- Performing Test BLAS_HAS_CBLAS - Failed
-- Performing Test BLAS_HAS_LAPACKE
-- Performing Test BLAS_HAS_LAPACKE - Failed
CMake Error at /usr/share/cmake-3.16/Modules/FindPackageHandleStandardArgs.cmake:146 (message):
  Could NOT find LAPACKE (missing: LAPACKE_CBLAS_LIB)
Call Stack (most recent call first):
  /usr/share/cmake-3.16/Modules/FindPackageHandleStandardArgs.cmake:393 (_FPHSA_FAILURE_MESSAGE)
  /home/joseph/opt/dplasma-termdet/share/cmake/dplasma/FindLAPACKE.cmake:313 (find_package_handle_standard_args)
  /home/joseph/opt/dplasma-termdet/share/cmake/dplasma/dplasma-config.cmake:51 (find_package)
  cmake/modules/FindDPLASMA.cmake:1 (find_package)
  CMakeLists.txt:85 (include)
```
even though `MKLROOT` is set correctly:
```
$ echo $MKLROOT 
/opt/intel/oneapi/mkl/2021.3.0
```

Is that an issue in DPLASMA?

--- a/cmake/modules/FindOrFetchBTAS.cmake
+++ b/cmake/modules/FindOrFetchBTAS.cmake
@@ -7,7 +7,12 @@ endif (NOT TARGET BTAS::BTAS)
 
 if (NOT TARGET BTAS::BTAS)
 
-  set(BTAS_ENABLE_MKL ${ENABLE_MKL})
+  # BTAS will load BLAS++/LAPACK++ ... if those use CMake's FindBLAS/FindLAPACK (as indicated by defined BLA_VENDOR)
+  # will need to specify Fortran linkage convention ... manually for now, switching to NWX's linear algebra discovery
+  # is necessary to handle all the corner cases for automatic discovery
+  if (DEFINED BLA_VENDOR)
+    set(_linalgpp_use_standard_linalg_kits TRUE)
+  endif(DEFINED BLA_VENDOR)
 
   FetchContent_Declare(
       BTAS
@@ -27,6 +32,26 @@ if (NOT TARGET BTAS::BTAS)
 
   # set BTAS_CONFIG to the install location so that we know where to find it
   set(BTAS_CONFIG ${CMAKE_INSTALL_PREFIX}/${BTAS_INSTALL_CMAKEDIR}/btas-config.cmake)
+
+  # define macros specifying Fortran mangling convention, if necessary
+  if (_linalgpp_use_standard_linalg_kits)
+    if (NOT TARGET blaspp AND NOT TARGET lapackpp)
+      message(FATAL_ERROR "ouch")
+    endif(NOT TARGET blaspp AND NOT TARGET lapackpp)
+    if (TTG_LINALG_MANGLING STREQUAL lower)
+      target_compile_definitions(blaspp PUBLIC -DBLAS_FORTRAN_LOWER=1)
+      target_compile_definitions(lapackpp PUBLIC -DLAPACK_FORTRAN_LOWER=1)
+    elseif(TTG_LINALG_MANGLING STREQUAL UPPER OR TTG_LINALG_MANGLING STREQUAL upper)
+      target_compile_definitions(blaspp PUBLIC -DBLAS_FORTRAN_UPPER=1)
+      target_compile_definitions(lapackpp PUBLIC -DLAPACK_FORTRAN_UPPER=1)
+    else()
+      if (NOT TTG_LINALG_MANGLING STREQUAL lower_)
+        message(WARNING "Linear algebra libraries' mangling convention not specified; specify -DTTG_LINALG_MANGLING={lower,lower_,UPPER}, if needed; will assume lower_")
+      endif(NOT TTG_LINALG_MANGLING STREQUAL lower_)
+      target_compile_definitions(blaspp PUBLIC -DBLAS_FORTRAN_ADD_=1)
+      target_compile_definitions(lapackpp PUBLIC -DLAPACK_FORTRAN_ADD_=1)
+    endif()
+  endif (_linalgpp_use_standard_linalg_kits)
 
 endif(NOT TARGET BTAS::BTAS)
 

